### PR TITLE
Re-enable pkg-config tests on windows

### PR DESCRIPTION
**Pull request by h-vetinari**
I tried adding CMake metadata in conda-forge/libprotobuf-feedstock#128 (to re-enable the CMake tests on unix), but this caused some fall-out. However, CMake has its own custom `FindProtobuf` [function](https://github.com/Kitware/CMake/blob/v3.24.1/Modules/FindProtobuf.cmake) that takes precedence over the packaged CMake metadata; perhaps it's still possible to nudge it to find our `protobuf-config`-less protobuf)

### CODE REVIEW

**Comment by h-vetinari** [comment]
@traversaro
Any idea how to make CMake's custom `FindProtobuf` work also on unix? I've tried setting a bunch of variables I see being used in the function, but so far to no avail...

**Comment by traversaro** [comment]
> @traversaro Any idea how to make CMake's custom `FindProtobuf` work also on unix? I've tried setting a bunch of variables I see being used in the function, but so far to no avail...

The Protobuf CMake package is searched for in https://github.com/grpc/grpc/blob/v1.46.4/cmake/protobuf.cmake#L58 : 
~~~
find_package(Protobuf REQUIRED ${gRPC_PROTOBUF_PACKAGE_TYPE})
~~~

if `gRPC_PROTOBUF_PACKAGE_TYPE` is `CONFIG`, then the use of `FindProtobuf.cmake` is skipped, and the CMake only searches for `ProtobufConfig.cmake` or `protobuf-config.cmake` (so-called `find_package` config mode, see https://cmake.org/cmake/help/v3.24/command/find_package.html). So we should avoid that `gRPC_PROTOBUF_PACKAGE_TYPE` is set to `CONFIG`. Based on https://github.com/grpc/grpc/blob/v1.46.4/CMakeLists.txt#L89, I guess the other possible value for  `gRPC_PROTOBUF_PACKAGE_TYPE`  is `MODULE`, that even if it is not an option for `find_package`, at least it is ignored so `find_package` will go back to the default behaviour of using `FindProtobuf.cmake`.

**TL;DR**: pass `-DgRPC_PROTOBUF_PACKAGE_TYPE:STRING="MODULE"` to CMake config.

**Comment by h-vetinari** [comment]
Thanks a lot, I'll try this shortly!

**Comment by h-vetinari** [comment]
> Thanks a lot, I'll try this shortly!

No success, unfortunately... 😑
